src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,65 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ *  Represents a specific type of Task that has a deadline as additional information.
+ *  Following the convention for String input is crucial for successful instantiation.
+ */
+public class Deadline extends Task {
+    protected String by;
+    protected LocalDate date;
+    protected LocalTime time;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by.trim();
+        String[] timeComponent = by.split(" ");
+        this.date = LocalDate.parse(format(timeComponent[0].trim()));
+        String hour = timeComponent[1].substring(0,2);
+        String minute = timeComponent[1].substring(2);
+        this.time = LocalTime.of(Integer.parseInt(hour),
```

Comment:
Another possible way to do this it to define a DateTimeFormatter pattern and directly parse with that pattern.